packages/pass-style/src/hex.js

```diff
@@ -0,0 +1,206 @@
+ * @deprecated Just use the exported `uint8ArrayToHex` and `hexToUint8Array`
+ * which are already best efforts based on feature detecting `Buffer`.
+ */
+export const makePortableHexCodec = () => {
+  /**
+   * @param {Uint8Array} u8
+   * @returns {string}
+   */
+  const uint8ArrayToHex = u8 => Array.from(u8, b => encodings[b]).join('');
+
+  /**
+   * @deprecated Use `uint8ArrayToHex` instead
+   * @param {Uint8Array} u8
+   * @returns {string}
+   */
+  const encodeHex = u8 => uint8ArrayToHex(u8);
+
+  /**
+   * @param {string} hex
+   * @returns {Uint8Array}
+   */
+  const hexToUint8Array = hex => {
+    const inputLen = hex.length;
+    if (inputLen % 2 !== 0) {
+      throw Fail`Invalid hex string: ${hex}`;
+    }
+    const u8 = new Uint8Array(inputLen / 2);
+    for (let i = 0; i < inputLen; i += 2) {
+      const b = getDecoding(hex.slice(i, i + 2));
+      // eslint-disable-next-line no-bitwise
+      u8[i >> 1] = b;
+    }
+    // TODO once we can rely on a native implementation of
+    // Immutable ArrayBuffer, do a transferToImmutable and return a new
+    // hardened Uint8Array on that buffer. Cannot yet because the
+    // Immutable ArrayBuffer shim does not support TypedArrays on
+    // emulated Immutable ArrayBuffers.
+    return harden(u8); // special case does not freeze index members.
+  };
+
+  /**
+   * @deprecated Use `hexToUint8Array` instead
+   * @param {string} hex
+   * @returns {Uint8Array}
+   */
+  const decodeHex = hex => hexToUint8Array(hex);
+
+  return harden({
+    uint8ArrayToHex,
+    encodeHex,
+    hexToUint8Array,
+    decodeHex,
+  });
+};
+harden(makePortableHexCodec);
+
+/**
+ * @typedef {Pick<BufferConstructor, 'from' | 'isBuffer'> & {
+ *   prototype: Pick<Buffer, 'toString'> & Uint8Array;
+ * }} BufferishConstructor
+ *   is the portion of the Node.js Buffer API we need for hex conversion.
+ */
+
+/**
+ * Create a hex codec using parts of the Node.js Buffer API.
+ *
+ * @deprecated Just use the exported `uint8ArrayToHex` and `hexToUint8Array`
+ * which are already best efforts based on feature detecting `Buffer`.
+ * @param {BufferishConstructor} Bufferish the object that implements the
+ *   necessary pieces of Buffer
+ * @returns {HexCodec}
+ */
+export const makeBufferishHexCodec = Bufferish => {
+  /**
+   * @param {Uint8Array} u8
+   * @returns {string}
+   */
+  const uint8ArrayToHex = u8 =>
+    (Bufferish.isBuffer?.(u8) ? u8 : Bufferish.from(u8)).toString('hex');
+
+  /**
+   * @deprecated Use `uint8ArrayToHex` instead
+   * @param {Uint8Array} u8
+   * @returns {string}
+   */
+  const encodeHex = u8 => uint8ArrayToHex(u8);
+
+  /**
+   * @param {string} hex
+   * @returns {Uint8Array}
+   */
+  const hexToUint8Array = hex => {
+    const buf = Bufferish.from(hex, 'hex');
+
+    // Coerce to Uint8Array to avoid leaking the abstraction.
+    const u8a = new Uint8Array(
+      buf.buffer,
+      buf.byteOffset,
+      buf.byteLength / Uint8Array.BYTES_PER_ELEMENT,
+    );
+    return u8a;
```

Comment:
I just noticed that the `makePortableHexCodec` `hexToUint8Array` hardens its output but the `makeBufferishHexCodec` analog does not, which is presumably an unintentional difference.

Reply:
Good catch. Yes, unintentional.